### PR TITLE
Move the core of qbft into a qbft core module

### PR DESCRIPTION
**Pull request by jframe**
## PR description
The first part to create a reusable QBFT library. This PR moves the core of QBFT into a new QBFT core module and keeps the Besu specific parts in the current QBFT module.

Further PRs will add QBFT specific types to the QBFT core module so it doesn't directly depend on the Besu block, block header and evm.

**Changes**
* Move the core packages from QBFT into a new QBFT core module. This leaves the blockcreation, header validation, protocol, jsonrpc and validator providers in the QBFT module
* Adapt the unit tests in the QBFT core module to use a mock of the BftExtraDataCodec so they don't depend on the QbftExtraDataCode in the QBFT module
* I've left the QBFT core integration tests to depend on the main QBFT module for now and will remove that in a follow-up PR. The change to remove this dependency was larger than I would like to include in this PR.

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: consensus/qbft-core/build.gradle [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline consensus/qbft-core/build.gradle:2]: nit prob should be Besu

**Review by macfarla — APPROVED** [review]
LGTM - maybe run the soak test?

**Comment by jframe** [comment]
> LGTM - maybe run the soak test?

Good idea, will run that before merging

**Review by Gabriel-Trintinalia — APPROVED** [review]
lgtm

**Pull request merged**